### DISCUSSION TITLE
default renovate.json

### DISCUSSION THREAD

**Discussion by tcurdt** [renovatebot/renovate#14250] [Request Help]
I am currently onboarding some repos. In every repo I change the default `renovate.json` to

```
{
  "extends": ["config:base", ":disableDependencyDashboard"]
}
```

Is there a way to tell renovate to create that kind of `renovate.json` in the first place?

**Comment by rarkins** [comment]
How are you running Renovate?

**Reply by tcurdt** [reply]
Doh! Sorry I forgot that "tiny" detail.
Not self-hosted with github.

**Reply by rarkins** [reply]
Do you mean you use the hosted app on github.com?

**Reply by tcurdt** [reply]
Yup. That's the one.

**Reply by HonkingGoose** [reply]
I think you want to create a "shareable config preset". Here's the documentation for [Shareable Config Presets](https://docs.renovatebot.com/config-presets/). 😉 

The big idea is that you have a `default.json` file in the `tcurd/renovate-config` repository.
Then you extend from that `default.json` config in all `renovate.json` files in all your other repositories.

Also read the section on [GitHub-hosted Presets](https://docs.renovatebot.com/config-presets/#github-hosted-presets) in the docs.

**Reply by tcurdt** [reply]
@HonkingGoose I looked at that - but the way I understand it I would still have to modify the original

```
{                
  "extends": [   "config:base" ]
}                
```

into

```
{
  "extends": ["github>rarkins/renovate-config"]
}
```

for every repo I add.

I am looking for a way to change the initial contents of the file.

**Reply by rarkins** [reply]
Please read https://docs.renovatebot.com/config-presets/#organization-level-presets

**Reply by tcurdt** [reply]
Thanks for the pointer.

From what I understand I just need to create a repo named `renovate-config` with `default.json`.
But I am a little lost on the 2nd paragraph or rather the example in combination with it.

> If found, that repository's preset will be suggested as the sole extended preset, and any existing onboardingConfig config will be ignored/overridden. For example the result may be:

```
{
  "$schema": "https://docs.renovatebot.com/renovate-schema.json",
  "extends": ["local>myorgname/.github:renovate-config"]
}
```

Can't I just create the `default.json` as

```
{
  "extends": ["config:base", ":disableDependencyDashboard"]
}
```

Or does that then have somehow higher priority (no `local>`) than what is inside the normal project repos?

**Reply by rarkins** [reply]
Yes, create the `default.json` exactly as you described.
